ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -646,9 +738,66 @@ impl Module {
                 return ConstIdx::new(idx);
             }
         }
-        // type table miss, we need to insert it.
+        // const table miss, we need to insert it.
         self.push_const(c.clone())
     }
+
+    /// Get the index of a type, inserting it into the type table if necessary.
+    pub(crate) fn type_idx(&mut self, t: &Type) -> Result<TypeIdx, CompilationError> {
+        // FIXME: can we optimise this?
+        for (idx, tt) in self.types.iter_enumerated() {
```

Comment:
Could this just be `self.types.position(t)` or `self.types.iter().position(t)`?

Reply:
Fixed in https://github.com/ykjit/yk/pull/982/commits/e63cc6dc772b0bd3b42775c4c22c4b6c503559c8

Fixed another in https://github.com/ykjit/yk/pull/982/commits/e63dd9041298a45c40656652db8d4b0c9e839fab, which should be a separate commit as it was pre-existing.